plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineConfigurationUtil.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.facets;
+
+import com.google.cloud.tools.eclipse.util.io.ResourceUtils;
+import java.io.InputStream;
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+
+/**
+ * Utility class for resolving App Engine configuration files in a project. The goal of this class
+ * is to insulate callers from having to know how and when configuration files are placed within a
+ * project. For example, Java projects for the flexible environment typically place their YAML
+ * configuration files in {@code src/main/appengine} whereas Java projects for the standard
+ * environment place their XML configuration files within the {@code WEB-INF} directory.
+ *
+ * @see WebProjectUtil
+ */
+public class AppEngineConfigurationUtil {
+  /** A well-known location for App Engine configuration files. */
+  private static final IPath DEFAULT_CONFIGURATION_FILE_LOCATION = new Path("src/main/appengine");
+
+  /**
+   * Create an App Engine configuration file in the appropriate location for the project.
+   *
+   * @param project the hosting project
+   * @param relativePath the file name
+   * @param contents the content for the file
+   * @param overwrite if {@code true} then overwrite the file if it exists
+   */
+  public static IFile createConfigurationFile(
+      IProject project,
+      IPath relativePath,
+      InputStream contents,
```

Comment:
I was going to say this could just be `String`, but then `WebProjectUtil.createFileInWebInf()` takes `InputStream` too. Looking at the callers, they could both accept `String`, but I don't want to stretch this PR for that. I just think this can just be `String`, what about you?